practices/git_flow.md

```diff
@@ -0,0 +1,77 @@
+
+## Why You Might Want to Use the Git Flow Workflow
+- Git Flow Workflow simplifies parallel development because it isolates the new development from the released project. You can work on any project version. The new development is done in feature branches so it is merged back when the developer is satisfied with the code.
+- Your team can collaborate better and spends less time managing the project versions if they use simple and clear branching strategy, such as Git Flow Workflow.
+- You can always commit the changes and create a new feature branch when you need to interrupt your work. You can come back to your feature at any time.
+- Hotfix branches allow your team to make emergency changes. You do not have to worry that you would accidentally merge in a new development at the same time.
+
+## Issues the Git Flow Workflow Solves
+- Unhappy clients
+- [Poor code quality](/issues/poor-code-quality)
+- Toxic team culture
+- Knowledge hoarding
+- Not my problem mentality
+- Meaningless work
+
+## How to Implement the Git Flow Workflow
+There are lots of different applications for Git Flow Workflow (such as [SmartGit](https://www.syntevo.com/smartgit/) or [GitKraken](https://www.gitkraken.com/). Pick the application suitable for your operation system to initialize the Git Flow. Customize your project setup by following these steps:
+1. Start using git-flow by cloning an existing git repository.
+2. Create an empty develop branch to complement the default master branch. Push it to the server.
+3. Development of new features starts from the develop branch. Each feature resides in its own branch. Push the feature branches to the git repository for backup.
+4. Merge the feature branch back into develop branch when the feature is finished.
+5. Create a release branch when the features are ready for a release. The parent is the develop branch.
+6. Merge the release branch into the master branch and tag it with a version number. After that, it is merged back into the develop branch so the progress since the release is saved. The release branch will be deleted.
+7. If you need to make some critical changes, create a hotfix branch. Fork the branch directly off of the master branch. When you are finished with the changes, merge the hotfix branch to both master and develop branch (or the current release branch). Tag the master branch with an updated version number.
+
+## Common Pitfalls of the Git Flow Workflow
+- Not everyone in the team uses Git Flow Workflow. It is crucial for smooth project development to ensure that all the team understands and uses the same workflow.
+- The Git Flow Workflow is customized for a specific product. The setting is too complicated for a new developer.
+- The team has too many features and is unable to finish them before the release. Development is drifting away from production.
```

Comment:
Tady by bylo dobré zmínit jak to ovlivňuje Git Flow

Reply:
@pamichalik What do you think?

Reply:
regarding the text: "... team has too many features and is unable to finish them ..." I just wonder how this affects git flow?

Reply:
I meant that git flow affects the team when there are too many features. The team is not able to finish all of the features before the release.

Reply:
@Kristynaslomkova Could you add it to the article, please?

Reply:
OK, now I understand